low-level/files/opt/seagate/sspl/bin/consuldump.py

```diff
@@ -42,8 +42,12 @@ def __init__(self, localtion=os.getcwd(), dir_prefix="", existing=False, name=No
                 print(f'Error[{connerr}] consul connection refused Retry Index {retry_index}')
                 time.sleep(WAIT_BEFORE_RETRY)
             except Exception as gerr:
-                print(f'Error[{gerr}] consul error')
-                break
+                if 'no cluster leader' in gerr.lower():
+                    print(f'Error[{gerr}] consul connection refused Retry Index {retry_index}')
+                    time.sleep(WAIT_BEFORE_RETRY)
```

Comment:
In that case, we are not giving SSPL a chance to do some pre-requisite things to make a connection (by giving some sleep). This may lead to a condition that our retry index will exceed before connection and we will not be able to connect to consul.

Reply:
i meant Code could have been much optimized. And do what we need to do, like sleep,  before 'continue'. Or keep just condition to break the loop alone (which is in else currently). Anyways, consider for future

Reply:
Ok. Will keep a TODO there.